python/lsst/meas/algorithms/computeExPsf.py

```diff
@@ -0,0 +1,305 @@
+        default=13579,
+    )
+
+    def validate(self):
+        # Docs inherited from base class
+        super().validate()
+        req_params = (self.nbins, self.bin_size, self.min_sep, self.max_sep)
+        num_req_params = sum(param is not None for param in req_params)
+        if num_req_params != 3:
+            msg = (
+                "You must specify exactly three of ``nbins``, ``bin_size``, ``min_sep`` and ``max_sep``"
+                f" in treecorr_config. {num_req_params} parameters were set instead."
+            )
+            raise FieldValidationError(self.__class__.bin_size, self, msg)
+
+        if self.min_sep is not None and self.max_sep is not None:
+            if self.min_sep > self.max_sep:
+                raise FieldValidationError(
+                    self.__class__.min_sep, self, "min_sep must be <= max_sep"
+                )
+
+
+class ComputeExPsfTask(Task):
+    """Compute Ex for PSF.
+
+    Compute scalar correlation function from
+    PSF ellipticity residuals to compute TEx
+    metrics.
+
+    Parameters
+    ----------
+    de1: `np.ndarray`
+        PSF ellipticity residuals component 1.
+    de2: `np.ndarray`
+        PSF ellipticity residuals component 2.
+    ra: `np.ndarray`
+        Right ascension coordinate.
+    dec: `np.ndarray`
+        Declination coordinate.
+    units: `str`
+        In which units are ra and dec. units supported
+        are the same as the one in treecorr.
+
+    Returns
+    -------
+    struct : `lsst.pipe.base.Struct`
+            The struct contains the following data:
+        ``E1``: `float`
+            <de1 de1> scalar correlation function, compute
+            in an angular bin define in TreecorrConfig.
+        ``E2``: `float`
+            <de2 de2> scalar correlation function, compute
+            in an angular bin define in TreecorrConfig.
+        ``Ex``: `float`
+            <de1 de2> scalar cross-correlation function, compute
+            in an angular bin define in TreecorrConfig.
+    """
+
+    ConfigClass = TreecorrConfig  # ComputeExPsfConfig
+    _DefaultName = "computeExPsf"
+
+    def run(
+        self,
+        de1: npt.NDArray,
+        de2: npt.NDArray,
+        ra: npt.NDArray,
+        dec: npt.NDArray,
+        units: str = "arcmin",
+    ) -> pipeBase.Struct:
+
+        kwargs_cat = {
+            "ra": ra,
+            "dec": dec,
+            "ra_units": units,
+            "dec_units": units,
+        }
+
+        cat1 = treecorr.Catalog(k=de1, **kwargs_cat)
+        cat2 = treecorr.Catalog(k=de2, **kwargs_cat)
+
+        config_kk = self.config.toDict()
+
+        kk = treecorr.KKCorrelation(config_kk)
+
+        kk.process(cat1)
+        kk_E1 = copy.deepcopy(kk.xi[0])
+        kk.process(cat2)
+        kk_E2 = copy.deepcopy(kk.xi[0])
+        kk.process(cat1, cat2)
+        kk_Ex = copy.deepcopy(kk.xi[0])
+
+        return pipeBase.Struct(E1=kk_E1, E2=kk_E2, Ex=kk_Ex)
```

Comment:
There is some awkwardness in starting a variable with upper case (as opposed to a class). I don't have a better suggestion at the moment.

Reply:
I guess this is just because in all tickets TEx is upper case and I guess putting lower case can introduce confusion with ellipticity  parameters. Open to suggestions and don't have a big feeling on this.

Reply:
One option would be to call them `metric_E1` instead of `E1`.